packages/transit-vehicle-overlay/src/utils.js

```diff
@@ -0,0 +1,298 @@
+export function checkHeading(heading) {
+  if (heading === null || heading < 0 || heading >= 360) {
+    heading = 0;
+  }
+  return heading;
+}
+
+/**
+ * Checks if a parameter is actually a function.
+ * @param {*} fn The function to call.
+ * @returns fn if fn is a function, or a dummy function.
+ * TODO: copied from map ... should be in core-utils?
+ */
+export function callIfValid(fn) {
+  if (typeof fn === "function") return fn;
+  return () => {};
+}
+
+/**
+ * geojson uses [lon,lat] (e.g., [X, Y]) in representing coordinates
+ * this utility function reverses the point order to be [lat, lon] (or [Y, X])
+ *
+ * @return array of reversed points in the line geom
+ */
+export function reverseGeojsonPointsInGeom(geom) {
+  const revPoints = [];
+  for (let i = 0; i < geom.coordinates.length; i++) {
+    const c = geom.coordinates[i].reverse();
+    revPoints.push(c);
+  }
+  return revPoints;
+}
+
+/**
+ * OTP encodes polylines - this method will decode such geometries
+ * @param geom
+ * @return decoded polyline
+ */
+export function decodePolyline(geom) {
+  let retVal = geom;
+  if (geom && geom.points) {
+    retVal = polyline.decode(geom.points);
+  }
+  return retVal;
+}
+
+/**
+ * find a vehicle based on an id (either vehicle or trip id) within a list of vehicles
+ * @param vehicleList - list of vehciles to be searched
+ * @param queryId - either trip or vehcile id (string) to search list
+ * @param defVal - return value when not found or errors happen
+ */
+export function findVehicleById(vehicleList, queryId, defVal = null) {
+  let retVal = defVal;
+  try {
+    if (vehicleList && queryId) {
+      vehicleList.some(v => {
+        if (queryId === v.vehicleId || queryId === v.tripId) {
+          retVal = v;
+          return true;
+        }
+        return false;
+      });
+    }
+  } catch (e) {
+    console.log(e);
+  }
+  return retVal;
+}
+
+/**
+ * do 2 vehicle have the same tracking ids
+ * @param vehicleA
+ * @param vehicleB
+ * @returns boolean
+ */
+export function isTracked(vehicleA, vehicleB) {
+  let retVal = false;
+
+  try {
+    if (vehicleA && vehicleB) {
+      if (
+        vehicleA.vehicleId === vehicleB.vehicleId ||
+        vehicleA.tripId === vehicleB.tripId
+      ) {
+        retVal = true;
+      }
+    }
+  } catch (e) {
+    console.log(e);
+  }
+  return retVal;
+}
+
+/** build a url from sub parts */
+export function buildUrl(base, query) {
+  let retVal = base;
+  try {
+    retVal = `${base}/${query}`.replace(/([^:]\/)\/+/g, "$1");
+  } catch (e) {
+    retVal = `${base}/${query}`;
+  }
+  return retVal;
+}
+
+/**
+ * query real-time vehicles, and return the results in the set* parameter callbacks
+ *
+ * @param setData - callback where vehicles array will be passed
+ * @param url - url of the vehicle service
+ */
+export function fetchVehicles(setData, trackedId, baseUrl, query) {
+  // build url
+  baseUrl = baseUrl || "https://maps.trimet.org/gtfs/rt/vehicles/";
```

Comment:
I think we should try to avoid TriMet-specific config and urls as much as possible. The component that calls this already has a default set, so there's probably no need to have a fallback value again here.